src/whylogs/core/statistics/stringtracker.py

```diff
@@ -0,0 +1,258 @@
+import logging
+import math
+
+from typing import List 
+
+from datasketches import frequent_strings_sketch
+
+from whylogs.core.statistics.thetasketch import ThetaSketch
+from whylogs.core.summaryconverters import from_string_sketch
+from whylogs.proto import CharPosMessage, CharPosSummary, StringsMessage, StringsSummary
+from whylogs.util import dsketch
+from whylogs.util.util_functions import encode_to_integers
+
+from .numbertracker import NumberTracker
+
+MAX_ITEMS_SIZE = 128
+MAX_SUMMARY_ITEMS = 100
+
+logger = logging.getLogger(__name__)
+
+
+class CharPosTracker:
+    def __init__(self, character_list: str = "abcdefghijklmnopqrstuvwzyz0123456789-@!#$%^&*()[]{\}"):
+
+        self.character_list = set(character_list)
+        self.char_pos_map = {}
+
+    def update(self, value):
+        for indx, char in enumerate(value.lower()):
+
+            try:
+                char = char.encode("ascii")
+                if char in self.character_list:
+                    self.char_pos_map.setdefault(char, NumberTracker())
+                    self.char_pos_map[char].track(indx)
+                else:
+                    self.char_pos_map.setdefault("NITL", NumberTracker())
+                    self.char_pos_map["NITL"].track(indx)
+            except UnicodeEncodeError:
+                self.char_pos_map.setdefault("NITL", NumberTracker())
+                self.char_pos_map["NITL"].track(indx)
+
+    def merge(self, other):
+        """
+        Merges two Char Pos Frequency Maps
+
+        Args:
+
+            other (CharPosTracker): to be merged
+
+        """
+        new_character_list = self.character_list.union(other.character_list)
```

Comment:
this look wrong. Should you attempt to de-dupe the characters so that "abc" union with "abc" doesn't result in "abcabc"?